src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -501,33 +584,141 @@ internal override void OnItemClicked(int clickedIndex)
 						// The CollectionView may have intercepted the change
 						clickedIndex = collectionView.CurrentPosition;
 					}
-					SelectedIndex = clickedIndex;
+
+					if (modifiers.HasFlag(VirtualKeyModifiers.Control) && clickedIndex == SelectedIndex)
+					{
+						SelectedIndex = -1;
+					}
+					else
+					{
+						SelectedIndex = clickedIndex;
+					}
+
 					break;
 				case ListViewSelectionMode.Multiple:
+					if (!modifiers.HasFlag(VirtualKeyModifiers.Shift))
+					{
+						FlipSelectionInMultipleSelection(clickedIndex, item: clickedItem, container: clickedContainer);
+						break;
+					}
+
+					// if shift is held, all items between the clicked item and the previously
+					// focused item match the selection of the previously focused item
+					var newSelection = focusedContainer?.IsSelected ?? false;
```

Comment:
does the focused changed during shift clicking?
from a quick experiment, i noted the following with a listview of 5 items: item0~item4
- (no mod) click item-2 // select item-2
- shift click item-0 // select item-0 to item-2
- shift click item-4 // select item-2 to item-4
- shift click item-2 // does nothing
does `focusContainer` remain the same for you during this scenario?

Reply:
i didnt mean item-0 to item-1 were unselected, the comment only the change since last step
this is done in `Multiple` mode
so based from this observation, we see that: 

> shift-click an item always cause a range action (select or unselected, based on the last no-mod click result)
> from the shift-click target to last no-mod-click target.

the question is do you maintain this (last no-mod-click target),
or does it change after each click?

Reply:
Oh, sorry. In my implementation, the focusedContainer changes on any click (even shift clicking). I'm not sure if it matters if shift-click changes focusedContainer since shift-clicking makes both (the old and the new) focused containers either selected on unselected. Is there a specific scenario where this would break things?

Reply:
It matters in the sense that the shift clicking always apply un/selection 
first) between 1st and last clicked items,
second) and not between the second last and last clicked items.
And, the fact that `FocusedIndexContainerItem` gets updated on each click, suggests that LV would use the 2nd behavior.

Reply:
Aren't both behaviors actually the same? The end result will be that all items between MinIndex(1st, second last, last) and MaxIndex(1st, second last, last) will be selected, right? It doesn't matter if the second shift click (un)selects between 1st and last shift-clicked items or between second last and last shift-clicked items because in the second scenario, the items between 1st and second last clicked items are already (un)selected?